src/core.js

```diff
@@ -440,7 +440,7 @@ JSONEditor.prototype = {
       callback();
     }
   },
-  expandRefs: function(schema) {
+  expandRefs: function(schema, recurseAllOf) {
```

Comment:
@sirockin Have you checked if any of the other instances of `expandRef` benefits from adding the extra Boolean parameter? (Note: the line numbers might not match the version you are using)

**\json-editor\src\core.js           3 matches**
[56:25]	      var schema = self.expandRefs(self.schema);
[443:3]	  expandRefs: function(schema) {
[545:17]	    return this.expandRefs(extended);

**\json-editor\src\editors\array.js           3 matches**
[181:35]	        var tmp = this.jsoneditor.expandRefs(this.schema.items);
[220:30]	    schema = this.jsoneditor.expandRefs(schema);
[234:30]	    schema = this.jsoneditor.expandRefs(schema);

**\json-editor\src\editors\multiple.js           1 matches**
[87:32]	      schema = self.jsoneditor.expandRefs(schema);

**\json-editor\src\editors\multiselect.js           1 matches**
[34:40]	        items_schema = this.jsoneditor.expandRefs(this.schema.items || {}),

**\json-editor\src\editors\table.js           1 matches**
[22:39]	    var item_schema = this.jsoneditor.expandRefs(this.schema.items || {});

**\json-editor\src\validator.js           1 matches**
[20:41]	    schema = $extend({},this.jsoneditor.expandRefs(schema));

Reply:
Not yet, no. I haven't completely got my head around the interactions going on here so I'm nervous about breaking any existing functionality. I'm also on a bit of a deadline with the work I'm doing right now. 

How confident are you that the e2e tests cover existing functionality? It's always reassuring to see tests pass after changing code :-)

I have a feeling I'll be working on this again so will keep in mind your comments and look at the other invocations next time I dig in there.

Reply:
I can't see any problems with applying your patch to all instances of `expandRefs`. If the functionality is currently broken. 
And `oneOf`, `anyOf` and `allOf` all "share" the same functionality, so the other 2 probably have same problems.
As for the tests, then I don't think they are 100%. We discover issues not covered by the tests from time to time. 😄